components/tools/OmeroJava/test/integration/AbstractServerTest.java

```diff
@@ -285,6 +298,155 @@ public void tearDown() throws Exception {
     }
 
     /**
+     * Cast the map of strings (e.g. the ALL_GROUPS_CONTEXT)
+     * into implicit context, which asks for <String, String>
+     * @param implicitContext the implicit context to be changed
+     * @param newContext the map of strings (e.g. ALL_GROUPS_CONTEXT)
+     */
+    protected void mergeIntoContext(ImplicitContext implicitContext, Map<String, String> newContext) {
+        for (final Entry<String, String> entry : newContext.entrySet()) {
+            implicitContext.put(entry.getKey(), entry.getValue());
+        }
+    }
+
+    /**
+     * Add the given annotation to the given image.
+     * @param image an image
+     * @param annotation an annotation
+     * @return the new loaded link from the image to the annotation
+     * @throws ServerError an error possibly occurring during saving of the link
+     */
+    protected ImageAnnotationLink linkImageAnnotation(Image image, Annotation annotation) throws ServerError {
```

Comment:
Barring the fact that starting to use the gateway in the integration tests will make it harder to extract the gateway to its own jar, having integration tests at least start to use the gateway matches the current Python strategy. In fact, treating those client-side classes of `blitz/` similarly to `tools/OmeroPy` may be the cleanest way to go about decoupling.